python/lsst/ctrl/mpexec/cli/opt/optionGroups.py

```diff
@@ -0,0 +1,123 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+__all__ = ("butler_options", "execution_options", "meta_info_options", "pipeline_build_options",
+           "qgraph_options")
+
+
+from functools import partial
+
+from lsst.daf.butler.cli.utils import option_section, split_kv, unwrap
+import lsst.obs.base.cli.opt as obsBaseOpts
+import lsst.daf.butler.cli.opt as dafButlerOpts
+from . import options as ctrlMpExecOpts
+
+instrumentOptionHelp = ("Add an instrument which will be used to load config overrides when defining a "
+                        "pipeline. This must be the fully qualified class name.")
+
+
+class OptionGroup():
+    """Base class for an option group decorator. Requires the option group
+    subclass to have a property called `decorator`."""
+
+    def __call__(self, f):
+        for decorator in reversed(self.decorators):
+            f = decorator(f)
+        return f
+
+
+class pipeline_build_options(OptionGroup):  # noqa: N801
+
+    def __init__(self):
+        self.decorators = [
+            option_section(sectionText="Pipeline build options:"),
+            ctrlMpExecOpts.pipeline_option(),
+            ctrlMpExecOpts.task_option(),
+            ctrlMpExecOpts.delete_option(metavar="LABEL"),
+            dafButlerOpts.config_option(metavar="LABEL:NAME=VALUE", multiple=True),
+            dafButlerOpts.config_file_option(help=unwrap("""Configuration override file(s), applies to a task
+                                                         with a given label."""),
+                                             metavar="LABEL:FILE",
+                                             multiple=True),
+            ctrlMpExecOpts.order_pipeline_option(),
+            ctrlMpExecOpts.save_pipeline_option(),
+            ctrlMpExecOpts.pipeline_dot_option(),
+            obsBaseOpts.instrument_option(help=instrumentOptionHelp, metavar="instrument", multiple=True)]
+
+
+class qgraph_options(OptionGroup):  # noqa: N801
+
+    def __init__(self):
+        self.decorators = [option_section(sectionText="Quantum graph building options:"),
+                           ctrlMpExecOpts.qgraph_option(),
+                           ctrlMpExecOpts.skip_existing_option(),
+                           ctrlMpExecOpts.save_qgraph_option(),
+                           ctrlMpExecOpts.save_single_quanta_option(),
+                           ctrlMpExecOpts.qgraph_dot_option()]
```

Comment:
Nitpick - there is a mixture of wrapping/indentation styles in different classes like

    self.decorators = [decorator1,
                       decorator2]

vs.

    self.decorators = [
        decorator1,
        decorator2
    ]

I personally prefer latter style, feels more readable as it lets you pack longer lines without more wrapping.

Reply:
Oh, latter is nice. 👍